### ISSUE TITLE
Error while populate is running for the `summ-example`

### ISSUE THREAD

**Issue by ursuad** [yasyf/summ#3]
Hi, I'm just running the `summ-example` command. 
If I click on the screen while the populate is running (on the interface and on the name of the text files on the left), I get the following error and the program crashes: 

```
summ-example
Starting up, please hold...
ACTION select_cursor DirectoryTree(pseudo_classes={'focus', 'focus-within', 'hover'})
ACTION select_cursor DirectoryTree(pseudo_classes={'focus', 'focus-within', 'hover'})
ACTION select_cursor DirectoryTree(pseudo_classes={'focus', 'focus-within', 'hover'})
╭───────────────────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────────────────────────────────────────────────╮
│ /opt/homebrew/Cellar/summ-lib/0.1.14/libexec/lib/python3.11/site-packages/summ/cli/widgets/output_tree.py:84 in _on_node                                                                                     │
│                                                                                                                                                                                                              │
│    81 │                                                                                                                                                                                                      │
│    82 │   def _on_node(self, entry: Node):                                                                                                                                                                   │
│    83 │   │   self.log(entry)                                                                                                                                                                                │
│ ❱  84 │   │   self.screen.post_message_no_wait(self.UpdateTokens(self, Chain.tokens_used()))                                                                                                                 │
│    85 │   │                                                                                                                                                                                                  │
│    86 │   │   if entry.parent:                                                                                                                                                                               │
│    87 │   │   │   parent = self.nodes.get(entry.parent)                                                                                                                                                      │
│                                                                                                                                                                                                              │
│ ╭────────────────────────────────────────────────────────────────────────────────────── locals ──────────────────────────────────────────────────────────────────────────────────────╮                       │
│ │ entry = Entry(                                                                                                                                                                     │                       │
│ │         │   thread=1204907,                                                                                                                                                        │                       │
│ │         │   color='magenta',                                                                                                                                                       │                       │
│ │         │   indent=1,                                                                                                                                                              │                       │
│ │         │   title='Document [?/12]',                                                                                                                                               │                       │
│ │         │   text='c2534',                                                                                                                                                          │                       │
│ │         │   parent=Entry(                                                                                                                                                          │                       │
│ │         │   │   thread=1204907,                                                                                                                                                    │                       │
│ │         │   │   color=None,                                                                                                                                                        │                       │
│ │         │   │   indent=2,                                                                                                                                                          │                       │
│ │         │   │   title='',                                                                                                                                                          │                       │
│ │         │   │   text='Counted, a sea lion, is undergoing a delicate brain surgery. The surgeons and nu'+173,                                                                       │                       │
│ │         │   │   parent=None                                                                                                                                                        │                       │
│ │         │   )                                                                                                                                                                      │                       │
│ │         )                                                                                                                                                                          │                       │
│ │  self = OutputTree(id='output-tree')                                                                                                                                               │                       │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                       │
│                                                                                                                                                                                                              │
│ /opt/homebrew/Cellar/summ-lib/0.1.14/libexec/lib/python3.11/site-packages/textual/dom.py:358 in screen                                                                                                       │
│                                                                                                                                                                                                              │
│   355 │   │   while node is not None and not isinstance(node, Screen):                         ╭───────────────── locals ─────────────────╮                                                                  │
│   356 │   │   │   node = node._parent                                                          │   node = None                            │                                                                  │
│   357 │   │   if not isinstance(node, Screen):                                                 │ Screen = <class 'textual.screen.Screen'> │                                                                  │
│ ❱ 358 │   │   │   raise NoScreen("node has no screen")                                         │   self = OutputTree(id='output-tree')    │                                                                  │
│   359 │   │   return node                                                                      ╰──────────────────────────────────────────╯                                                                  │
│   360 │                                                                                                                                                                                                      │
│   361 │   @property                                                                                                                                                                                          │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
NoScreen: node has no screen
```

**Comment by yasyf** [comment]
Looking into it!

**Comment by yasyf** [comment]
@ursuad what version of python are you using?

**Comment by francisverhoeven** [comment]
I'm having the same issue @yasyf.

python --version returns Python 2.7.16
python3 --version returns Python 3.8.9

**Comment by francisverhoeven** [comment]
Actually, after lots of updating & upgrading, I'm getting a different error.

pyenv is set to 3.11.2

Running "summ-example", everything works until I click the "populate" button. Then it loads for 50%, and returns the following error:

<img width="944" alt="CleanShot 2023-03-10 at 00 11 51@2x" src="https://user-images.githubusercontent.com/29704283/224181615-0d203667-ab26-4e40-848c-cf3454e92c36.png">

![CleanShot 2023-03-10 at 00 17 06@2x](https://user-images.githubusercontent.com/29704283/224182269-6b8875bb-a6fe-40e9-a810-ba1ea8109681.png)

Would really love to try this out (code isn't my strong suit unfortunately)

**Comment by jeffcatania** [comment]
I am getting the same error. I suspect that the current `main` branch / `v0.1.14` is broken.

[The Screen class ](https://github.com/yasyf/summ/blob/34fac727330a85de0e9c781a4e4494b638d1e7bf/summ/cli/screens/screen.py#L10)does not have a `post_message_no_wait` method.

Similarly, if I comment out the erroneous line, I get a new error.  This error points to the calling function not having the same number of positional arguments as expected. 
<img width="1242" alt="image" src="https://user-images.githubusercontent.com/355412/225398240-22a370e5-db07-4d53-a9e7-34c5194fc074.png">

Is there a stable version that I can revert to?

**Comment by jeffcatania** [comment]
Following up from my previous post, it looks like textual has updated its API in version 0.14 to drop the ‘post_no_wait’ method. My current suspicion is that this error is caused by incorrect dependency versioning. 

https://textual.textualize.io/blog/2023/03/09/textual-0140-shakes-up-posting-messages/

**Comment by yasyf** [comment]
@jeffcatania ah, that'll do it. I'll pin the dependency

**Comment by yasyf** [comment]
@willmcgugan breaking changes without a major version bump! grumble grumble 😛 

https://github.com/Textualize/textual/pull/1940

**Comment by willmcgugan** [comment]
Textual is zero ver, so it is best to pin or keep a close eye on the release notes...

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

**Comment by yasyf** [comment]
@willmcgugan fair enough! will do :)

**Comment by yasyf** [comment]
@jeffcatania should be fixed in `0.1.16`!
